Carnap/src/Carnap/Languages/PurePropositional/Logic/Ripley.hs

```diff
@@ -55,9 +57,13 @@ instance Inference RipleyLNJ PurePropLexicon (Form Bool) where
         ruleOf DisjElim   = proofByCasesVariations !! 0
         ruleOf As         = axiom        
 
-        indirectInference x
-           | x `elem` [CondIntro, DisjElim] = Just PolyProof
-           | otherwise = Nothing
+        globalRestriction (Left ded) n CondIntro = Just (dischargeConstraint n ded (view lhs $ conclusionOf CondElim))
+        globalRestriction (Left ded) n DisjElim = Just (dischargeConstraint n ded (view lhs $ conclusionOf DisjElim))
+        globalRestriction _ _ _ = Nothing
+
+        indirectInference CondIntro = Just $ TypedProof (ProofType 1 1)
```

Comment:
`indirectInference` basically tells you what kind of indirect inference you're dealing with. `TypedProof (ProofType 1 1)` means that this is an indirect proof with one assumption, where only the last line of the subproof is relevant to the inference rule. In the Lemmon context, this means that you're discharging one dependency, and the number of relevant final lines is just ignored. 

`TypedProof (ProofType 2 _)` means you're discharging two constraints. 

If you had, e.g. a reductio rule in a fitch system that assumes ψ and has to end with two lines matching φ,¬φ, then that'd be a `TypedProof (ProofType 1 2)`.